sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/HiveTypeCoercion.scala

```diff
@@ -324,6 +324,7 @@ trait HiveTypeCoercion {
    *   e1 union e2  max(s1, s2) + max(p1-s1, p2-s2)         max(s1, s2)
    *   sum(e1)      p1 + 10                                 s1
    *   avg(e1)      p1 + 4                                  s1 + 4
+   *   compare      max(p1, p2)                             max(s1, s2)
```

Comment:
This line should be moved elsewhere and reworded, since comparisons result in boolean values rather than decimal values.